### PR TITLE
Quantum Volume: Add the ability to specify the device graph to compile the model circuit onto

### PR DESCRIPTION
**Pull request by KevinVillela**
This should give us the ability to try different parts of the device to find the optimal configuration. Note that the qubits that were used are already returned in the QuantumVolumeResult.compiled_circuit property.

@mpharrigan

### CODE REVIEW

**Comment by KevinVillela** [comment]
Filed https://github.com/quantumlib/Cirq/issues/2646 for the Travis failure.

File: cirq/contrib/quantum_volume/quantum_volume.py [hunk 1]
```diff
@@ -473,9 +474,12 @@ def calculate_quantum_volume(
                                 depth=depth,
                                 num_circuits=num_circuits,
                                 random_state=random_state)
+    device_graph = ccr.gridqubits_to_graph_device(device_or_qubits) if type(
```
Comment (mpharrigan) [inline cirq/contrib/quantum_volume/quantum_volume.py]: python nit:

```python
if isintance(device_graph, list):
  device_graph = blah
else:
  device_graph = blah2
```

Comment (KevinVillela) [reply cirq/contrib/quantum_volume/quantum_volume.py]: Done

**Comment by KevinVillela** [comment]
Windows pytest failed...
Wtf is going on today haha

File: cirq/contrib/quantum_volume/quantum_volume.py [hunk 2]
```diff
@@ -475,9 +476,14 @@ def calculate_quantum_volume(
                                 depth=depth,
                                 num_circuits=num_circuits,
                                 random_state=random_state)
-    device_graph = ccr.gridqubits_to_graph_device(device_or_qubits) if type(
-        device_or_qubits) is list else ccr.xmon_device_to_graph(
-            device_or_qubits)
+
+    # Get the device graph from the given qubits or device.
+    device_graph = None
```
Comment (mpharrigan) [inline cirq/contrib/quantum_volume/quantum_volume.py]: thing of interest: usually you don't have to include this. most linters are smart enough to notice that `device_graph` is initialized no matter what and all the scoping in python is so horribly leaky you don't need to declare it outside

Comment (KevinVillela) [reply cirq/contrib/quantum_volume/quantum_volume.py]: mypy was complaining that the variable device_graph was referenced before assignment. I wasn't able to find an existing issue for this online, maybe its WAI?

Comment (mpharrigan) [reply cirq/contrib/quantum_volume/quantum_volume.py]: ah interesting

**Pull request merged**